### PR TITLE
Updated WriteCheckpoint API

### PR DESCRIPTION
**Pull request by jeongkyun-oh**
## Proposed changes

- WriteCheckpoint API is used to update checkpoint manually. After updating, new checkpoint is stored in database but not in memory. This PR updates the API and makes it only when fetching is stopped.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: datasync/chaindatafetcher/api.go [hunk 1]
```diff
@@ -51,6 +56,15 @@ func (api *PublicChainDataFetcherAPI) ReadCheckpoint() (int64, error) {
 }
 
 func (api *PublicChainDataFetcherAPI) WriteCheckpoint(checkpoint int64) error {
+	isRunning := atomic.LoadUint32(&api.f.fetchingStarted)
+	if isRunning == running {
+		return errors.New("please stopFetching first before write checkpoint manually")
```
Comment (yoomee1313) [inline datasync/chaindatafetcher/api.go:61]:
```suggestion
		return errors.New("please call stopFetching first before write checkpoint manually")
```

Comment (jeongkyun-oh) [reply datasync/chaindatafetcher/api.go]: Reflected :) Thanks.

**Review by yoomee1313 — APPROVED** [review]

**Review by aidan-kwon — DISMISSED** [review]

**Review by hyochan-brown — APPROVED** [review]

File: datasync/chaindatafetcher/api.go [hunk 2]
```diff
@@ -51,6 +56,12 @@ func (api *PublicChainDataFetcherAPI) ReadCheckpoint() (int64, error) {
 }
 
 func (api *PublicChainDataFetcherAPI) WriteCheckpoint(checkpoint int64) error {
+	isRunning := atomic.LoadUint32(&api.f.fetchingStarted)
+	if isRunning == running {
+		return errors.New("please call stopFetching first before write checkpoint manually")
+	}
+
+	api.f.checkpoint = checkpoint
```
Comment (ehnuje) [inline datasync/chaindatafetcher/api.go:64]: Is there any possibility to concurrent write of `checkpoint`?

Comment (jeongkyun-oh) [reply datasync/chaindatafetcher/api.go:64]: @ehnuje `checkpoint` is updated only when `fetching` is started which is checked the above statement. Apart from that, it is not possible to access `checkpoint` concurrently.

File: datasync/chaindatafetcher/api.go [hunk 3]
```diff
@@ -51,6 +56,12 @@ func (api *PublicChainDataFetcherAPI) ReadCheckpoint() (int64, error) {
 }
 
 func (api *PublicChainDataFetcherAPI) WriteCheckpoint(checkpoint int64) error {
+	isRunning := atomic.LoadUint32(&api.f.fetchingStarted)
+	if isRunning == running {
+		return errors.New("please call stopFetching first before write checkpoint manually")
```
Comment (winnie-byun) [inline datasync/chaindatafetcher/api.go:61]:
```suggestion
		return errors.New("call stopFetching before writing checkpoint manually")
```

**Review by winnie-byun — APPROVED** [review]

File: datasync/chaindatafetcher/api.go (hunk 3 above)
Comment (jeongkyun-oh) [reply datasync/chaindatafetcher/api.go]: Thanks :)

**Review by winnie-byun — APPROVED** [review]

**Review by ehnuje — APPROVED** [review]

**Review by yoomee1313 — APPROVED** [review]

**Pull request merged**